job.go

```diff
@@ -24,8 +24,12 @@ type internalJob struct {
 	name   string
 	tags   []string
 	jobSchedule
-	lastScheduledRun   time.Time
-	nextScheduled      time.Time
+	lastScheduledRun time.Time
+
+	// as some jobs may queue up, it's possible to
+	// have multiple nextScheduled times
+	nextScheduled []time.Time
```

Comment:
If a job is running over and singleton mode is enabled, the job at the front of this list will get skipped. Do you plan to pop it off the front like a queue? If so, using a queue would be a nice abstraction here.